### PR TITLE
Entry activity layout design

### LINKED ISSUE #4: Configuration Page
**Describe the solution you'd like**
When I open the app I should see a screen which let's me configure the following things:

- The time I wake up
- The Time I go to sleep
- How many glasses of water I want to drink during the day
- How do I want to see the alarms. (notification, fullscreen alarm)
- I can choose 3 states for the sound (mute, Vibration, Alarm sound)

**Additional context**

- **To achieve this we need an additional class which is called the ViewModel.**
_The project must always follow the SOLID principles and MVVM architecture.
For understanding how the architecture is supposed to be you follow this [link](https://developer.android.com/jetpack/guide).
For better understanding of the MVVM term you can follow this [link](https://www.journaldev.com/20292/android-mvvm-design-pattern)._

- **A class is needed which is responsible for storing the configuration.**
_Remember the class is not bound to any specific activity. The configuration class must always be available to all the classes_
- **When the save button is pressed, the configuration must be saved.**

### PR DESCRIPTION
**Pull request by nrazavi70**

### CODE REVIEW

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_entry)
+        val waterReminderNotifGroup = findViewById<RadioGroup>(R.id.waterReminderNotifGroup)
+        val notificationSoundGroup = findViewById<RadioGroup>(R.id.notificationSoundGroup)
+        val entryPageSubmitButton = findViewById<Button>(R.id.entryPageSubmitButton)
+        val wakeUpTimeError = findViewById<TextView>(R.id.wakeUpTimeError)
+        val sleepTimeError = findViewById<TextView>(R.id.sleepTimeError)
+        val sharedPref = getSharedPreferences("EntryPageAnswers", Context.MODE_PRIVATE)
+        val sharedPrefEditor = sharedPref.edit()
```
Comment (saeedntt) [inline app/src/main/java/com/reminder/hydration/EntryActivity.kt]: Create a class for managing shared preferences. This class will be responsible for saving to and reading from shared preferences.

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        entryPageSubmitButton.setOnClickListener {
+            arrayOf(
+                findViewById<EditText>(R.id.wakeUpTimeH),
+                findViewById<EditText>(R.id.sleepTimeH),
+                findViewById<EditText>(R.id.drunkenWaterAmount),
+                findViewById<RadioButton>(waterReminderNotifGroup.checkedRadioButtonId),
+                findViewById<RadioButton>(notificationSoundGroup.checkedRadioButtonId)
```
Comment (saeedntt) [inline app/src/main/java/com/reminder/hydration/EntryActivity.kt]: If you are not executing the same command on all of the items no point in putting them inside an array.

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_entry)
+        val waterReminderNotifGroup = findViewById<RadioGroup>(R.id.waterReminderNotifGroup)
+        val notificationSoundGroup = findViewById<RadioGroup>(R.id.notificationSoundGroup)
+        val entryPageSubmitButton = findViewById<Button>(R.id.entryPageSubmitButton)
+        val wakeUpTimeError = findViewById<TextView>(R.id.wakeUpTimeError)
+        val sleepTimeError = findViewById<TextView>(R.id.sleepTimeError)
+        val sharedPref = getSharedPreferences("EntryPageAnswers", Context.MODE_PRIVATE)
+        val sharedPrefEditor = sharedPref.edit()
+        entryPageSubmitButton.setOnClickListener {
```
Comment (saeedntt) [inline app/src/main/java/com/reminder/hydration/EntryActivity.kt]: On click listeners are events. Events are the responsibility of the viewModel. Create a class with a proper name that will handle the events of the view and is responsible for view actions. Then create an instance of that class inside the activity. Then when you find the view inside activity, you pass it to the viewmodel. Then view model will take care of the actions related to that item.

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            arrayOf(
+                findViewById<EditText>(R.id.wakeUpTimeH),
+                findViewById<EditText>(R.id.sleepTimeH),
+                findViewById<EditText>(R.id.drunkenWaterAmount),
+                findViewById<RadioButton>(waterReminderNotifGroup.checkedRadioButtonId),
+                findViewById<RadioButton>(notificationSoundGroup.checkedRadioButtonId)
+            ).apply {
+                if ((this[0].text.toString().toInt() > 24) or (this[0].text.toString().toInt() < 0)) {
+                    wakeUpTimeError.text = "Please enter a valid time."
```
Comment (saeedntt) [inline app/src/main/java/com/reminder/hydration/EntryActivity.kt]: Use the TextInputLayout with TextInputEditText then you dont need to create an error text manually.

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                } else if ((this[1].text.toString().toInt() > 24) or (this[1].text.toString().toInt() < 0)) {
+                    sleepTimeError.text = "Please enter a valid time."
```
Comment (saeedntt) [inline app/src/main/java/com/reminder/hydration/EntryActivity.kt]: Same as above about TextInputLayout.

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                } else {
+                    for (key in 0..this.size - 1) {
+                        sharedPrefEditor.putString(key.toString(), this[key].text.toString())
```
Comment (saeedntt) [inline app/src/main/java/com/reminder/hydration/EntryActivity.kt]: Create proper keys. Indexes are not keys. You need to create a unique key for values that you want to add in shared preferences. Create a class that keeps your keys as static constants then you can use them in your view model to save meaningful values in shared preferences.

**Review by saeedntt — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt (hunk 2 above)
Comment (nrazavi70) [reply app/src/main/java/com/reminder/hydration/EntryActivity.kt]: I am executing the putString method of shared preferences on all of them. Is that enough reason to put them all in an array?

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt (hunk 4 above)
Comment (saeedntt) [reply app/src/main/java/com/reminder/hydration/EntryActivity.kt]: https://material.io/develop/android/components/text-fields

File: app/src/main/java/com/reminder/hydration/EntryActivity.kt (hunk 1 above)
Comment (nrazavi70) [reply app/src/main/java/com/reminder/hydration/EntryActivity.kt]: I tried creating a class with these variables for start but getSharedPreferences is "Unresolved Reference". I can't seem to be finding a way around it.
Also it seems redundant cause getSharedPreferences on its own is a class to handle shared preferences. Would you mind elaborating on what the purpose of this class is?

**Pull request closed without merging**